### PR TITLE
Give repository owners more visibility into their history

### LINKED ISSUE #216: Include repository source of logs in logs
https://elixirforum.com/t/logger-categorized-by-supervision-subtrees/6127/1

### PR DESCRIPTION
**Pull request by notriddle**
This adds a "History" tab, where crash reports and past batches can be seen

This will log crash reports (so this PR fixes #216, though none of that data exists yet) and allows all the information about past batches (which we have been keeping, BTW, so this tab should already be populated for most long-standing repos).

Next step: tackle #227

### CODE REVIEW

File: apps/bors_database/lib/crash.ex [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  Builds a changeset based on the `struct` and `params`.
+  """
+  def changeset(struct, params \\ %{}) do
+    struct
+    |> cast(params, [:crash, :component])
+  end
+
+  def all_for_project(project_id) do
+    from c in Crash,
+      where: c.project_id == ^project_id
```
Comment (Macarse) [inline apps/bors_database/lib/crash.ex]: is the `^` actually needed?

File: apps/bors_frontend/web/controllers/project_controller.ex [hunk 2]
```diff
@@ -89,6 +90,25 @@ defmodule BorsNG.ProjectController do
       update_branches: Project.changeset_branches(project)
   end
 
+  def log(_, :ro, _, _), do: raise BorsNG.PermissionDeniedError
```
Comment (Macarse) [inline apps/bors_frontend/web/controllers/project_controller.ex]: `:ro` is not allowed and `:rw` is? Sounds it should be the other way around

File: apps/bors_database/lib/crash.ex (hunk 1 above)
Comment (notriddle) [inline apps/bors_database/lib/crash.ex]: Yeah, it's part of the Ecto syntax. If you don't put it there, it'll complain about `project_id` not being declared within the query (and suggest using the `^` syntax if it's a local variable that you want to interpolate).

File: apps/bors_frontend/web/controllers/project_controller.ex (hunk 2 above)
Comment (notriddle) [inline apps/bors_frontend/web/controllers/project_controller.ex]: `:rw` is a superset of `:ro`. I picked the more stringent permission mostly to be conservative.

File: apps/bors_database/lib/crash.ex (hunk 1 above)
Comment (notriddle) [inline apps/bors_database/lib/crash.ex]: https://hexdocs.pm/ecto/Ecto.Query.html gives a description of the syntax for the `from` macro.

**Comment by notriddle** [comment]
Since you said it was approved, I'll go with that.

bors r= @Macarse

**Pull request merged**